src/AggregateFunctions/AggregateFunctionSparkbar.h

```diff
@@ -211,10 +221,22 @@ class AggregateFunctionSparkbar final
 
         for (auto & y : histogram)
         {
+            constexpr auto bucket_num = static_cast<Y>(BAR_LEVELS - 1);
+
             if (isNaN(y) || y <= 0)
+            {
                 y = 0;
+                continue;
+            }
+
+            /// handle potential overflow
+            if (y_max > bucket_num && y >= std::numeric_limits<Y>::max() / bucket_num)
+                y = y / (y_max / bucket_num);
             else
-                y = y * 7 / y_max + 1;
+                y = y * bucket_num / y_max;
+
+            if (y < std::numeric_limits<Y>::max())
```

Comment:
y is always <= bucket_num and bucket_num == 7, so this condition seems to always be true